Add to Migration, Syntax, and Contribution Guides

bmorelli25 opened the pull request:
### Summary

* Adds to migration, syntax, and contribution guides
* Adds issue templates for bugs and enhancements
* Adds scripts helpful for migration tasks

bmorelli25 commented on docs/source/migration/scripts/attribute-converter/attr.py:
```diff
@@ -0,0 +1,59 @@
+import re
```
@elastic/docs-engineering I wonder if there's somewhere better to put these scripts instead of in the docs dir?

Mpdreamz replied:
@bmorelli25 can you describe a bit what the intended use case is for these ?

Who runs them and when? is this a continuous thing?

bmorelli25 replied:
Good question. They're documented in the source files in this PR. But a brief summary is:

1. **CODEOWNER list for documentation freeze**
  This script creates a list of repositories and the directories in those repositories that the `@docs-freeze-team` (just @KOTungseth) should be added as codeowner of. During a documentation freeze, the `@docs-freeze-team` must approve any pull requests that modify `.asciidoc` files before they can be merged. This prevents folks from merging docs changes during the freeze.

1. **Attribute converter**
  This script converts the attributes in https://github.com/elastic/docs/blob/master/shared/attributes.asciidoc to the supported yaml format for docs-builder.

1. **HTML to Markdown Tab Converter**
  The tab converter script is used to convert HTML tabs into Markdown tabs within documentation files. The script scans through specified directories, finds all Markdown (`.md`) files, and replaces any HTML-based tab structures with their Markdown equivalents.<br>  
  _This is no longer necessary as @siamakp-elastic added this capability to the migration tool. I will remove this._

Mpdreamz replied:
1. Is this something that needs to run continuously in the future or is this a one of thing? 


2. Same question really, but also:
   * Do we need all of these in the new architecture?
   * Rather then having these live in all of the isolated `docset.yml` files should `docs-builder` natively support them?

bmorelli25 replied:
> Is this something that needs to run continuously in the future or is this a one of thing?

Okay so after a meeting today, the CODEOWNER script won't be used so we can get rid of it. We will need to rerun the attribute converter eventually (Jan 29) but I can do that so there's no need to put it here (won't be used after that date). I'll remove both from this PR.

> Do we need all of these in the new architecture?

All, no. Most, yes.

> Rather then having these live in all of the isolated docset.yml files should docs-builder natively support them?

Yes. The more I play with this, the more a single yml file where these can be defined would be ideal.

bmorelli25 commented:
The build is currently failing with

```
Error: Not linked in toc: migration/scripts/codeowner-gen/output.md
```

Opened a bug for this: https://github.com/elastic/docs-builder/issues/169

bmorelli25 commented:
As discussed above, I removed the scripts from this PR. That also fixed the build error. I'm going to get this merged so that the GH pages is updated.
